Sources/Tentacle/Tree.swift

```diff
@@ -81,6 +143,47 @@ public struct Tree: CustomStringConvertible {
 
         /// The mode of the entry.
         public let mode: Mode
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            self.sha = try container.decode(SHA.self, forKey: .sha)
+            self.path = try container.decode(String.self, forKey: .path)
+            self.mode = try container.decode(Mode.self, forKey: .mode)
+            self.type = try EntryType(from: decoder)
+        }
+
+        public init(type: EntryType, sha: SHA, path: String, mode: Mode) {
+            self.type = type
+            self.sha = sha
+            self.path = path
+            self.mode = mode
+        }
+
+        private enum CodingKeys: String, CodingKey {
+            case type
+            case sha
+            case path
+            case mode
+            case url
+            case size
+        }
```

Comment:
This shouldn't be needed since all the cases have the default value?

Reply:
If think they are because url and size are not exposed as properties and we are using them for encoding, but maybe I'm missing something..?